benches/serde.rs

```diff
@@ -413,78 +393,60 @@ struct ValueAfterTable {
     qux: bool,
 }
 
-generate_benches!(
-    (array, Array),
-    (basic, Basic),
-    (basic_empty_let, Basic),
-    (boolean, Boolean),
-    (chained, Chained),
-    (chained_complex, ChainedComplex),
-    (char, Char),
-    (comment, Basic),
-    (compact, Compact),
-    (complex, Complex),
-    (complex_keys, ComplexKeys),
-    (environment_variable, Basic),
-    (float, Float),
-    (input, Input),
-    (input_references_input, Basic),
-    (integer, Integer),
-    (mixed_array, MixedArray),
-    (null, Null),
-    (object, Object),
-    (object_in_array, ObjectInArray),
-    (readme_example, ReadmeExample),
-    (string, String_),
-    (string_interpolation, Basic),
-    (value_after_table, ValueAfterTable),
-    (very_compact, Compact) // (basic_new_type_enum, BasicNewTypeEnum),
-                            // (basic_unit_enum, BasicUnitEnum),
-                            // (basic_new_type, BasicNewType),
-                            // (bytes, Bytes),
-                            // (chained_enum, ChainedEnum),
-                            // (mixed_array_enum, MixedArrayEnum),
-                            // (null_option, NullOption),
-                            // (null_unit, NullUnit),
-                            // (str, Str)
-);
+macro_rules! bench_case {
+    ($group:expr, $name:literal, $input_file:literal, $output_type:ty) => {
+        let input = include_str!(concat!("../assets/inputs/", $input_file, ".corn"));
+        $group.throughput(Throughput::Bytes(input.len() as u64));
+        $group.bench_function($name, |b| {
+            b.iter(|| {
+                let result: Result<$output_type, _> = corn::from_str(black_box(input));
+                black_box(result.unwrap())
+            })
+        });
+    };
+}
```

Comment:
If this macro were inside the function, the `group` arg could be dropped.

Reply:
I intentionally wrote it with external hygiene so that in the future we can reuse it if we have more benchmarking groups. I feel it's prob good to leave it as is but it's an easy change if you'd prefer it that way.